src/libraries/Common/src/System/Text/ValueStringBuilder_1.cs

```diff
@@ -0,0 +1,314 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Buffers;
+using System.Diagnostics;
+using System.Runtime.CompilerServices;
+using System.Runtime.InteropServices;
+
+#nullable enable
+
+namespace System.Text
+{
+    internal ref partial struct ValueStringBuilder<TChar>
+        where TChar : unmanaged
+    {
+        private TChar[]? _arrayToReturnToPool;
+        private Span<TChar> _chars;
+        private int _pos;
+
+        public ValueStringBuilder(Span<TChar> initialBuffer)
+        {
+            Debug.Assert((typeof(TChar) == typeof(Utf8Char)) || (typeof(TChar) == typeof(Utf16Char)));
```

Comment:
If this is specific to these types that are defined just in System.Runtime.Numerics, and it's only used by System.Runtime.Numerics, we probably shouldn't have the file in Common?

Reply:
Is `ValueStringBuilder<TChar>` actually needed? Its implementation looks almost the same as `ValueListBuilder<T>` .

Reply:
There's various differences in growth and other strategies based on the typical differences between building strings and building collections. I think it's still relevant to have both for the same reason we have StringBuilder and ImmutableArray.Builder/etc

Reply:
I understand that there are differences in strategies. However, as shown below, `System.Number` appears to take the approach of handling both `byte` and `char` with `ValueListBuilder<T>`, so adding `ValueStringBuilder<T>` seemed inconsistent to me.


https://github.com/dotnet/runtime/blob/290d04d4a5fb28b6939469151d60432f6899a631/src/libraries/System.Private.CoreLib/src/System/Number.Formatting.cs#L715-L726
https://github.com/dotnet/runtime/blob/290d04d4a5fb28b6939469151d60432f6899a631/src/libraries/System.Private.CoreLib/src/System/Number.Formatting.cs#L774-L785

Reply:
That's likely more historical then anything else. We should likely update it to use ValueStringBuilder for consistency and since we are actually building strings